DirectProgramming/DPC++FPGA/Tutorials/Features/ac_fixed/README.md

```diff
@@ -0,0 +1,291 @@
+
+To use an `ac_fixed` type in your code, include the following header:
+
+```cpp
+#include <sycl/ext/intel/ac_types/ac_fixed.hpp>
+```
+> *IMPORTANT*: You must pass the  `-qactypes` option on Linux or the `/Qactypes` option on Windows to the `dpcpp` command when compiling your SYCL program in order to ensure that the headers are correctly included. In this tutorial, the options are passed through the `src/CMakeLists.txt` file.
+
+### Recommended Method for Constructing `ac_fixed` Numbers
+
+The compiler uses significant FPGA resources to convert double precision (and single precision) floating-point values to `ac_fixed` values. The kernel `ConstructFromFloat` in the function `TestConstructFromFloat` constructs an `ac_fixed` object from an accessor to a native `float` type.
+
+In contrast, the kernel `ConstructFromACFixed` in the function `TestConstructFromACFixed` constructs an `ac_fixed` object from an accessor to another `ac_fixed` object. This consumes far less area than the previous kernel. See the section on *Examining the Reports* below to understand where to look for this difference within the optimization reports.
+
+### Using the `ac_fixed` Math Functions
+
+To use the `ac_fixed` math functions in your code, include the following header:
+
+```cpp
+#include <sycl/ext/intel/ac_types/ac_fixed_math.hpp>
+```
+
+The functions `TestCalculateWithFloat` and `TestCalculateWithACFixed` in this tutorial design contain the kernels `CalculateWithFloat` and `CalculateWithACFixed` respectively. Both calculate the simple expression:
+```cpp
+  square_root ( sine(x) * sine(x) + cosine(x) * cosine(x) )
+```
+for some input `x`.
+
+The kernel `CalculateWithFloat` uses floating point values and the standard math library while `CalculateWithACFixed` uses `ac_fixed` values and the `ac_fixed` math library.
+
+In the kernel `CalculateWithACFixed`, the `sin_fixed` and `cos_fixed` functions require the integer part's bit width to be 3, and the input value range to be within [-pi, pi]. Therefore, `ac_fixed` inputs are instantiated with the following parameters:
+
+```cpp
+  W = 10, I = 3, S = true
+```
+
+In this tutorial, the `ac_fixed` numbers are smaller in size than floating point numbers, which results in a reduction of the FPGA resources at the expense of accuracy. To see the trade-offs in accuracy, compare the numeric results of the operations. The area utilization differences will be discussed in the section on *Examining the Reports*.
+
+When you use the `ac_fixed` library, keep the following points in mind:
+
+- Input Bit Width and Input Value Range Limits
+
+    The fixed-point math functions have bit width and input value range requirements. All bit width and input value range requirements are documented at the top of the `ac_fixed_math.hpp` files.
```

Comment:
Are they also documented in the documentation linked above? Where is a user going to find this file?

Reply:
The document is in oneAPI optimization guide:  https://www.intel.com/content/www/us/en/develop/documentation/oneapi-fpga-optimization-guide/top/optimize-your-design/resource-use/data-types-and-operations/var-prec-fp-sup/declare-and-use-the-ac-data-types/declare-the-ac-fixed-data-type.html,

which also refers to the `ac_fixed_math.hpp` file. Should we just mention this page in the tutorial or directly refer to the header file?